core/gui/src/app/workspace/service/report-print/report-print.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from "@angular/core";
+import html2canvas from "html2canvas";
+import * as joint from "jointjs";
+import { WorkflowActionService } from "../workflow-graph/model/workflow-action.service";
+import { WorkflowResultService } from "../workflow-result/workflow-result.service";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ReportPrintService {
+  private paper!: joint.dia.Paper;
+  constructor(
+    public workflowActionService: WorkflowActionService,
+    private workflowResultService: WorkflowResultService
+  ) {}
+
+  /**
+   * Captures the current workflow snapshot and returns it as a base64-encoded PNG image URL.
+   * @returns {Promise<string>} A promise that resolves with the base64-encoded PNG image URL of the workflow snapshot.
+   */
+  public async getWorkflowSnapshot(): Promise<string> {
```

Comment:
Avoid use `async` and `Promise`, as it is blocking the UI. Use `Observe<string>` as the return type. And the caller of the method should subscribe for the result.

For how to use `Observe` and how to subscribe this kinds of return value, you can checkout other service